core/trino-main/src/main/java/io/trino/connector/ConnectorName.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorName
+{
+    private static final Pattern VALID_NAME = Pattern.compile("[a-z][a-z0-9_]*");
```

Comment:
This is a breaking change. There are plenty of connectors that have `-` in it out there. Until recently even Trino had `hive-hadoop2` connector. Why `-` is disallowed?

Reply:
probably because that makes it required to quote the name in SQL .. same as in session properties where we use _ instead of - 

And fyi .. even if this is a breaking change.. I agree with doing that to keep it cleaner.

Reply:
This is not a breaking change.  The code that reads the end-user's input file converts dashes to underscores, so this has no impact on existing installations.